src/utils/setImmediate.js

```diff
@@ -1,3 +1,4 @@
-const setImmediate = typeof global !== 'undefined' && typeof global.setImmediate !== 'undefined' ? global.setImmediate : (fn, ms) => setTimeout(fn, ms)
+const hasNativeSupport = typeof global !== 'undefined' && typeof global.setImmediate !== 'undefined'
+const setImmediate = hasNativeSupport ? (fn, ms) => global.setImmediate : (fn, ms) => setTimeout(fn, ms)
```

Comment:
returned functions returns `global.setImmediate` instead of calling it with provided arguments ;)